.test-infra/metrics/grafana/dashboards/perftests_metrics/Python_Performance_Tests.json

```diff
@@ -0,0 +1,297 @@
+{
```

Comment:
- @kamilwu , could you please review the dashboard config?
Some questions:
- Is there a way to visualize the dashboard on an in-progress PR?
- Is it possible to reduce duplication in the configs? We can do it in a separate change. For example: most settings for Python 2.7, and 3.7 are similar.  Have we considered  auto-generating  the final config from a smaller set of settings?

Reply:
@tvalentyn 

> Is there a way to visualize the dashboard on an in-progress PR?

Currently, the best way is to: pull branch, deploy metrics stack locally (`docker-compose build`; `docker-compose up`), open web browser and go to localhost:3000.

> Is it possible to reduce duplication in the configs?

There's a concept called [Scripted Dashboards](https://grafana.com/docs/grafana/latest/reference/scripting/), but we didn't implement it. So, if we want to create two dashboards next to each other, we have to provide two configs for them (even if they are very similar). 

Apart from that, we can also consider another question: is having those .json files version-controlled profitable? Diffs are often very large, so the standard review process does not apply here.

Reply:
@piotr-szuberski 

Could you rename your dashboard and charts? `Python Performance Tests` and `Python27/37 Performance Tests | 1GB` doesn't say much. How about using `Python WordCount IT Benchmarks` from the old Perfkit? [1]

[1] https://apache-beam-testing.appspot.com/explore?dashboard=5691127080419328

Reply:
Python WordCount IT Benchmarks definitely sounds better.

Reply:
Good catch. 

Re: 

> Apart from that, we can also consider another question: is having those .json files version-controlled profitable? Diffs are often very large, so the standard review process does not apply here.

I agree that those files are hard to review - I was thinking that perhaps we can autogenerate the final configs from a much smaller config that just captures the minimal necessary information (name of the benchmark to display,  table/row name in the database that has the datapoints). The idea is that final configs are not edited manually. It would make sense for the visualization tool to provide a capability to generate the dashboard based on a common template + required minimal config. Is that what Scripted Dashboards is? The webpage page says this feature is deprecated.

Reply:
> The idea is that final configs are not edited manually.

Let me explain: the configs are not edited manually. At least they shouldn't be. The current solution involves creating a local instance of Grafana, exporting a new dashboard to JSON file, and pasting that file into the git repo. I found an interesting discussion on Grafana's repo: https://github.com/grafana/grafana/issues/13823. It looks like this workflow is quite popular among users.

> Is that what Scripted Dashboards is?

Probably. But I guess we'd rather avoid using deprecated features.

Reply:
>  exporting a new dashboard to JSON file

Thanks, does this mean: creating a new dashboard manually via UI?

Reply:
Yes, you're right.

Reply:
Thanks - is this workflow documented somewhere on Beam's website? It could be a few pointers to Grafana's documentation,  maybe screenshots.

Reply:
There's an instruction on wiki: https://cwiki.apache.org/confluence/display/BEAM/Community+Metrics#CommunityMetrics-UpdatingDashboards.